test/testdata/lsp/code_actions/extract_class_func.rb

```diff
@@ -0,0 +1,36 @@
+# typed: strict
+# selective-apply-code-action: refactor.extract
+
+module Foo
+  extend T::Sig
+
+  sig {returns(String)}
+  def self.greeting
+         # ^^^^^^^^ apply-code-action: [A] Extract method to module
```

Comment:
I believe that we only get `DefinitionRespnse` results back for the `declLoc`, which applies to anywhere on the `def` line, but not the method body or the `end` keyword.

Do you mind adding some more tests, for e.g., the `def` keyword, the space after `def`, the `self` keyword, and the `.` token? I think that your code handles all of these case already, but they'd be good to capture in a test.

Also, I think that for these tests, the test harness only uses the first `^` in the `apply-code-action`, and all the remaining `^^^` after the first are ignored (fwiw).